NBitcoin.Altcoins/Viacoin.cs

```diff
@@ -70,17 +174,62 @@ public ViacoinBlock(ViacoinBlockHeader header) : base(header)
 			{
 
 			}
+
 			public override ConsensusFactory GetConsensusFactory()
 			{
 				return ViacoinConsensusFactory.Instance;
 			}
 		}
+		public class ViacoinBlockHeader : BlockHeader
+		{
+			const int VERSION_AUXPOW = (1 << 8);
 
+			AuxPow auxPow = new AuxPow();
+
+			public AuxPow AuxPow
+			{
+				get
+				{
+					return auxPow;
+				}
+				set
+				{
+					auxPow = value;
+				}
+			}
+
+			public override uint256 GetPoWHash()
+			{
+				var headerBytes = this.ToBytes();
+				var h = NBitcoin.Crypto.SCrypt.ComputeDerivedKey(headerBytes, headerBytes, 1024, 1, 1, null, 32);
+				return new uint256(h);
+			}
+
+			public override void ReadWrite(BitcoinStream stream)
+			{
+				base.ReadWrite(stream);
+				if((Version & VERSION_AUXPOW) != 0)
+				{
+					if(!stream.Serializing)
+					{
+						stream.ReadWrite(ref auxPow);
+					}
+				}
+			}
+		}
 #pragma warning restore CS0618 // Type or member is obsolete
 
-		protected override void PostInit()
+		//Format visual studio
+		//{({.*?}), (.*?)}
+		//Tuple.Create(new byte[]$1, $2)
+		//static Tuple<byte[], int>[] pnSeed6_main = null;
+		//static Tuple<byte[], int>[] pnSeed6_test = null;		
+
+		static uint256 GetPoWHash(BlockHeader header)
 		{
-			RegisterDefaultCookiePath("Viacoin");
```

Comment:
You would need this. This is used so that NBitcoin can connect to your local viacoin without configuration. I rely on this for BTCPay.